### PR TITLE
Add missing documentation in aster-frame: bus and device modules

### PR DESCRIPTION
**Pull request by sdww0**
This PR adds some missing documentation in the aster-frame crates. Related PR: #893

### CODE REVIEW

File: framework/aster-frame/src/bus/mmio/bus.rs [hunk 1]
```diff
@@ -35,6 +39,7 @@ pub struct MmioBus {
 }
 
 impl MmioBus {
+    /// Register MMIO driver to the MMIO bus.
```
Comment (StevenJiang1110) [inline framework/aster-frame/src/bus/mmio/bus.rs:42]: Register -> Registers

File: framework/aster-frame/src/bus/pci/capability/msix.rs [hunk 2]
```diff
@@ -147,11 +148,13 @@ impl CapabilityMsixData {
         }
     }
 
+    /// MSI-X Table size
     pub fn table_size(&self) -> u16 {
         // bit 10:0 table size
         (self.loc.read16(self.ptr + 2) & 0b11_1111_1111) + 1
     }
 
+    /// Enable a interrupt line, it will replace the old handle with the new handle.
```
Comment (StevenJiang1110) [inline framework/aster-frame/src/bus/pci/capability/msix.rs:157]: Enable -> Enables

File: framework/aster-frame/src/bus/pci/capability/vendor.rs [hunk 3]
```diff
@@ -23,41 +25,45 @@ impl CapabilityVndrData {
         }
     }
 
+    #[allow(clippy::len_without_is_empty)]
+    /// The length of this capability
```
Comment (StevenJiang1110) [inline framework/aster-frame/src/bus/pci/capability/vendor.rs:29]: The document shoule be above the attribute.

File: framework/aster-frame/src/bus/pci/cfg_space.rs [hunk 4]
```diff
@@ -1,5 +1,7 @@
 // SPDX-License-Identifier: MPL-2.0
 
+//! The PCI configuration space. Reference: [osdev](https://wiki.osdev.org/PCI)
+
```
Comment (StevenJiang1110) [inline framework/aster-frame/src/bus/pci/cfg_space.rs:4]: `Reference` should be in a new line. And add an empty line between Reference and the first statement. Other places should be refactored similarly.

```rust
//! The PCI configuration space. 
//! 
//! Reference: [osdev](https://wiki.osdev.org/PCI)
```

File: framework/aster-frame/src/bus/mmio/common_device.rs [hunk 5]
```diff
@@ -40,34 +42,43 @@ impl MmioCommonDevice {
...
     pub fn io_mem(&self) -> &IoMem {
         &self.io_mem
     }
 
+    /// Device ID
     pub fn device_id(&self) -> u32 {
         self.io_mem.read_val::<u32>(8).unwrap()
     }
 
+    /// Version of the MMIO device. FIXME: This function can only used in Virtio-mmio device.
```
Comment (StevenJiang1110) [inline framework/aster-frame/src/bus/mmio/common_device.rs:60]: FIXME should be in a new line

File: framework/aster-frame/src/bus/pci/cfg_space.rs [hunk 6]
```diff
@@ -12,74 +14,143 @@ use crate::{
...
         const FAST_BACK_TO_BACK_CAPABLE = 1 << 7;
+        /// This bit is only set when the following conditions are met:
+        /// 1. The bus agent asserted PERR# on a read or observed an assertion
+        /// of PERR# on a write
+        /// 2. The agent setting the bit acted as the bus master for the
+        /// operation in which the error occurred
+        /// 3. Bit 6 of the Command register (Parity Error Response bit) is set
+        ///  to 1.
         const MASTER_DATA_PARITY_ERROR  = 1 << 8;
+        /// Read only bits that represent the slowest time that a device will
```
Comment (StevenJiang1110) [inline framework/aster-frame/src/bus/pci/cfg_space.rs:122]: Read -> Reads. The starting verb in comments of this bitflags should be in a unified form, all with "s" or all without "s".

**Comment by tatetian** [comment]
@sdww0 Could you fix the conflict? After that, the PR would be good to get merged. Thanks!

**Comment by StanPlatinum** [comment]
A detailed comment with code also helps us to implement `sysfs` #957 . Thanks!

File: ostd/src/bus/mmio/common_device.rs [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+use super::VIRTIO_MMIO_MAGIC;
+use crate::{
+    io_mem::IoMem,
+    mm::{paddr_to_vaddr, Paddr, VmIo},
+    trap::IrqLine,
+};
+
+/// MMIO Common device.
+/// TODO: Implement universal access to MMIO devices since we are temporarily
+/// using specific virtio device as implementation of CommonDevice.
```
Comment (tatetian) [inline ostd/src/bus/mmio/common_device.rs:17]: TODO are almost always written for developers, not users. So TODOs should be written as comments, not Rust docs.

File: ostd/src/bus/mmio/common_device.rs [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+    }
+
+    /// Device ID
+    pub fn device_id(&self) -> u32 {
+        self.io_mem.read_val::<u32>(8).unwrap()
+    }
+
+    /// Version of the MMIO device.
+    ///
+    /// FIXME: This function can only used in Virtio-mmio devices.
```
Comment (tatetian) [inline ostd/src/bus/mmio/common_device.rs:62]: Same as TODOs. FIXMEs are almost always written for developers, not users. So FIXMEs should be written as comments, not Rust docs.

File: ostd/src/bus/pci/capability/msix.rs [hunk 9]
```diff
@@ -150,11 +152,13 @@ impl CapabilityMsixData {
         }
     }
 
+    /// MSI-X Table size
     pub fn table_size(&self) -> u16 {
         // bit 10:0 table size
         (self.loc.read16(self.ptr + 2) & 0b11_1111_1111) + 1
     }
 
+    /// Enables a interrupt line, it will replace the old handle with the new handle.
```
Comment (tatetian) [inline ostd/src/bus/pci/capability/msix.rs:161]: Typo: a --> an

**Review by tatetian — APPROVED** [review]
LGTM. Thanks for the contribution!

**Pull request merged**